### DISCUSSION TITLE
Detect USB scope disconnection and reconnect

### DISCUSSION THREAD

**Discussion by hernexto** [OpenHantek/OpenHantek6022#337] [Ideas]
Hi!
I have the 6022BL from many years ago and used the abandoned opensource version and now I find this version and it's AMAZING!!
Very good job! Thank you!

To the topic...
Seems my USB cable was dirty (or electronics needed a bit of warm) or loose (or whatever) but 6022 was disconnecting sometimes.
Of course OpenHantek freezes display and just restarting it works again.

So I wonder if OpenHantek could be able to detect disconnection and be able to reconnect.
Why reconnect if just restarting works?... well... anything that is able to autorecover ends helping somebody!
(I.E: not the way, but could helped in: Scope disconnects when trying to probe (6022BE) #166 )

Thank you!

**Comment by Ho-Ro** [comment]
> So I wonder if OpenHantek could be able to detect disconnection and be able to reconnect.
Why reconnect if just restarting works?... well... anything that is able to autorecover ends helping somebody!

A simple solution on the command line:

```sh
while true ; do OpenHantek && break ; done
```

Or as a shell script where you can provide options:

```sh
#!/bin/sh

# execute OpenHantek in a loop and restart
# if it was terminated by an error

while true; do
    OpenHantek $* && break
done
```

**Reply by hernexto** [reply]
Right!
Altrough I'm on your (not beloved) windows... it could be done the same way.

But I was wondering if something less tricky (aka automatic) could be done for inexperienced users.

Of course all that situation is not common so forget about this if it takes a lot of work to do it from inside app.

Thanks for listening!

**Reply by Ho-Ro** [reply]
I don't see any way to do this in a clean way, the error is triggered deep in the program by the usb library.
```
OpenHantek: ../../libusb/os/threads_posix.h:46: usbi_mutex_lock: Assertion `pthread_mutex_lock(mutex) == 0' failed.
```
I would need to restore state, free all dynamic structures, and do a bunch of other *housekeeping* tasks to avoid memory leaks. So the easiest way is a shutdown, which cleans everything up, and a fresh start. 
If you know the reason for the instability, wouldn't it be better to change the cable or fix the plug instead of programming around it?

**Reply by hernexto** [reply]
Then leave it as it is and focus on more useful things, this is not important at all!

If I wasn't clear, after cleaning USB contacts or using it a bit it stopped disconnecting, so I don't have this problem any more.
It was just an Idea, not a problem, bug or sth really needed. :)
